rfc/rfc006-distributed-registry.md

```diff
@@ -0,0 +1,146 @@
+# RFC006 Distributed Registry
+
+|  |  |
+| :--- | :--- |
+| Nuts foundation | R.G. Krul |
+| Request for Comments: 006 | Nedap |
+|  | September 2020 |
+
+## Distributed Registry
+
+### Abstract
+
+This RFC describes a protocol to build a registry containing information required for care organizations to exchange data
+through their software vendors. The registry typically contains vendors, organizations, services and endpoints.
+It uses [RFC004](rfc004-distributed-document-format.md)'s documents as underlying data format.
+
+### Status of document
+
+This document is currently a draft.
+
+### Copyright Notice
+
+![](../.gitbook/assets/license.png)
+
+This document is released under the [Attribution-ShareAlike 4.0 International \(CC BY-SA 4.0\) license](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## 1.  Introduction
+
+When care organizations want to exchange data using Nuts they need to know where to find that data and how to authenticate it.
+This RFC describes the document types containing the information required to form this registry.
+It uses [RFC004](rfc004-distributed-document-format.md) as underlying data format. The origin and requirements of
+certificates are defined by RFC Certificate Structure (TODO).
+
+## 2. Terminology
+
+* **Document**: a distributed document as specified by [RFC004](rfc004-distributed-document-format.md).
+* **Organization**: a care organization exchanging data with other care organizations over a Nuts network.
+* **Vendor**: an entity developing software for care organizations participating in a Nuts network. Organizations
+  access the Nuts network through their vendor's software.
+* **Endpoint**: typically a URI or URL exposed by a vendor which can be used by other vendors to pull data.
+* **Service**: a grouping of endpoints implementing a service required by a Bolt.
+
+Other terminology comes from the [RFC001 Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## 3. Updating
+To update an object (vendor, organization, endpoint or service) the vendor MUST create a new document with the same identifier
```

Comment:
Vendor identifier should be read from certificate since otherwise it's duplicate (both in certificate and in JSON). That leads to more validation logic (making sure they're the same).

Reply:
Yes, but it needs to be very explicit: the vendor document doesn't have an identifier field but it does have an extra validation based upon certificate contents.

Reply:
Added validation section